ui/src/VoxettaUserForm.js

```diff
@@ -187,69 +201,53 @@ export class VoxettaUserForm extends LitElement {
+     * Determines if each input in the user form is valid and alters Save Button appropriately. 
      */
-    ensureFormCompleteness() {
-        const idComplete = 
-            this.ensureInputCompleteness('user-id', 'Please enter a valid user id.');
-        const genderComplete =
-            this.ensureInputCompleteness('gender-list', 'Please select a gender.');
-        const ageComplete =
-            this.ensureInputCompleteness('user-age', 
-                'Please enter a valid age between 0 and 120.');
-        const deviceComplete =
-            this.ensureInputCompleteness('device-type', 'Please enter a device name.');
+    formIsValid() {
+        const userIdValidity = this.shadowRoot.getElementById('user-id').checkValidity();
+        const genderValidity = this.shadowRoot.getElementById('gender-list').checkValidity();
+        const userAgeValidity = this.shadowRoot.getElementById('user-age').checkValidity();
+        const deviceTypeValidity = this.shadowRoot.getElementById('device-type').checkValidity();
 
-        return idComplete && genderComplete && ageComplete && deviceComplete;  
+        const formValidity = userIdValidity && genderValidity && userAgeValidity && deviceTypeValidity;
+        
+        const saveButton = this.shadowRoot.getElementById('save-button');
+        formValidity ? saveButton.disabled  = false : saveButton.disabled = true; 
     }
 
     /**
-     * Ensures that a given input was complete when submitted. If not, it triggers an
-     * incompleteness-related error message for the given input. 
-     * @param {String} id - The HTML ID of the input whose completeness needs checked.
-     * @param {String} errMsg - The message to display if an input was 
-     *  submitted incomplete.
-     * @returns {Boolean} Denotes if a submitted input was complete.
+     * Emits an event that causes the form page to close and the record
+     * page to appear. 
      */
-    ensureInputCompleteness(id, errMsg) {
-        const input = this.shadowRoot.getElementById(id);
-        if (!input.value) {
-            input.setCustomValidity(errMsg);
-            input.reportValidity();
-            return false; 
-        } else {
-            input.setCustomValidity('');
-            input.reportValidity();
-            return true; 
-        }
+    handleExitForm() {
+        const event = new CustomEvent('exit-form', {});
+        this.dispatchEvent(event);
     }
 
     /**
-     * Emits an event that causes the form page to close and the record
-     * page to appear. 
+     * Emits an event that causes the user infomration to update. 
+     * @param {Object} userInfo - The information entered on the user form.
      */
-    handleExitForm() {
-        let event = new CustomEvent('exit-form', {
+    handleFormSubmission(userInfo) {
+        const event = new CustomEvent('update-user-info', {
             detail: {
-                state: 'RECORD-PAGE'
+                userInfo: userInfo
```

Comment:
you can write this simply

```javascript
detail: { userInfo }
```

If key is the same as the variable set as value, this applies. See shorthand object notation
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Object_initializer

Reply:
adjusted